File: members/U02BQ6G1SQJ/programmers/problem/신고 결과 받기.py
import collections;

def solution(id_list : list, report : list, k : int) -> list:
    answerList = [];
    idSet = set();
    reportDict = collections.defaultdict(set);
    reportCounter = collections.Counter();

    for curReport in report:
        (src, dst) = curReport.split(" ");

        if (dst not in reportDict[src]):
            reportDict[src].add(dst);
            reportCounter[dst] += 1;

            if (reportCounter[dst] >= k):
                idSet.add(dst);

    # print(idSet);
    # print(reportDict);
    # print(reportCounter);

    for curId in id_list:
        if (curId in reportDict.keys()):
            answerList.append(len(reportDict[curId] & idSet));
        else:
            answerList.append(0);

    return answerList;

# print(solution(["muzi", "frodo", "apeach", "neo"], ["muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"], 2)); # [2,1,1,0]
# print(solution(["con", "ryan"], ["ryan con", "ryan con", "ryan con", "ryan con"], 3));                                      # [0,0]

File: members/U02BWCHJA6Q/Programers/kakao_2022/신고_결과_받기.js
function solution(id_list, report, k) {
  const reported = new Map();
  const reportList = new Map();

  id_list.forEach((id) => {
    reported.set(id, new Set());
    reportList.set(id, new Set());
  });

  report.forEach((string) => {
    const [from, to] = string.split(' ');

    reported.get(to).add(from);
    reportList.get(from).add(to);
  });

  return id_list.map(
    (id) =>
      [...reportList.get(id)].filter((_id) => reported.get(_id).size >= k)
        .length
  );
}
